### DISCUSSION TITLE
Rendering on less fancy terminals

### DISCUSSION THREAD

**Discussion by burbma** [Textualize/textual#3799] [Q&A] [answered]
I'm making an app that I want to ship to an edge device. I'm using a raspberry pi 4 with their 7inch touchscreen for a display. I've gone for a minimal kiosk setup and so I have the device boot right into the app running via `startx` and `xterm` (with 256 colors). I've set some `.Xresources` to get it most of the way to looking as slick as it does on my dev machine (a mac running iterm) by using the same font (source code pro) and color theme (dracula). I have two images below, the first is a screenshot on my dev machine and the second is a picture I took of the app running on the pi. You'll notice that on the left edge of the one running on the pi there are some imperfections in the rendering showing as extra horizontal lines in the same color used to draw the box they extend from. I'm not very familiar with displaying things in terminals like this and am wondering if anyone has some pointers of things I could look into to crisp up the display in my minimal `xterm` setup.

<img width="317" alt="Screenshot 2023-12-02 at 11 11 20 AM" src="https://github.com/Textualize/textual/assets/7841889/a82e4953-790b-446d-847a-aa4f941a51b4">

![IMG_4982](https://github.com/Textualize/textual/assets/7841889/09053d4e-c10a-419e-a1b8-297860f63a11)

**Comment by davep** [comment]
You may find [this FAQ entry](https://textual.textualize.io/FAQ/#why-doesnt-textual-look-good-on-macos) useful in that is covers a similar issue in a different environment. My guess would be you would need to tweak the spacing between lines in that terminal emulator, or something similar.

**Comment by willmcgugan** [comment, marked as answer]
That glitch tends to occur when the line height doesn't match the height of the glyphs. If you can tweak the line height, you may be able to make it look as intended. If you can't, you may want to use different border types. Its the "wide" and "tall"   border types that suffer from this the most. If you avoid them, it will likely look better.

**Comment by burbma** [comment]
Thanks for the tips, I'll see what I can do. I'm enjoying the product and the community has been enjoyable!
